api-metastore/src/main/java/org/zalando/nakadi/service/SchemaEvolutionService.java

```diff
@@ -99,6 +107,48 @@ private void collectErrorMessages(final ValidationException e,
     public EventType evolve(final EventType original, final EventTypeBase eventType) throws SchemaEvolutionException {
         checkEvolutionIncompatibilities(original, eventType);
 
+        switch (eventType.getSchema().getType()) {
+            case JSON_SCHEMA:
+                return evolveJsonSchema(original, eventType);
+            case AVRO_SCHEMA:
+                return evolveAvroSchema(original, eventType);
+            default:
+                throw new RuntimeException("unsupported type for evolving " + eventType.getSchema().getType());
+        }
+    }
+
+    private EventType evolveAvroSchema(final EventType original, final EventTypeBase eventType) {
+        final var compatibilityMode = eventType.getCompatibilityMode();
+            validateAvroSchema(Collections.singletonList(original.getSchema()),
+                    eventType.getSchema(), compatibilityMode);
+
+        final var level = AvroUtils.getParsedSchema(original.getSchema().getSchema()).
+                equals(AvroUtils.getParsedSchema(eventType.getSchema().getSchema()))? NO_CHANGES : MAJOR;
+
+        return bumpVersion(original, eventType, level);
+    }
+
+    private void validateAvroSchema(final List<? extends EventTypeSchema> original,
```

Comment:
should indicate throwing of `SchemaEvolutionException`

Reply:
done